rdt/hyper_transformer.py

```diff
@@ -798,24 +798,30 @@ def _reverse_transform(self, data, prevent_subset):
             raise NotFittedError(self._NOT_FIT_MESSAGE)
 
         unknown_columns = self._subset(data.columns, self._output_columns, not_in=True)
+        if unknown_columns:
+            raise Error(
+                'There are unexpected column names in the data you are trying to transform. '
+                f'A reverse transform is not defined for {unknown_columns}.'
+            )
+
         if prevent_subset:
             contained = all(column in self._output_columns for column in data.columns)
             is_subset = contained and len(data.columns) < len(self._output_columns)
-            if unknown_columns or is_subset:
+            if is_subset:
                 raise Error(
                     'There are unexpected columns in the data you are trying to transform. '
```

Comment:
with the change you are making, idk if this error message makes sense anymore. There will no longer be unexpected columns here right? Just missing ones?